src/praisonai-agents/praisonaiagents/memory/adapters/factories.py

```
@@ -492,4 +502,15 @@ def _get_embedding(self, text: str) -> Optional[List[float]]:
             return result.embeddings[0] if result.embeddings else None
         except Exception:
-            return None
\ No newline at end of file
+            return None
+    
+    def close(self):
+        """Clean up MongoDB resources."""
+        if hasattr(self, 'client') and self.client:
+            try:
+                self.client.close()
+            except Exception as e:
```

Ruff (0.15.11)

[warning] Do not catch blind exception: `Exception`

(BLE001)